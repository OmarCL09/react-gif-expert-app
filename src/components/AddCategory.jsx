import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('One Punch Man');

    const onInputChange = ( e ) => {
        e.preventDefault();
        setInputValue( e.target.value );
    }

    const onSubmit = ( e ) => {
        e.preventDefault();
        if(inputValue.trim().length <= 1) return;
        setInputValue('');
        onAddCategory( inputValue.trim());
    }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
        <input 
            type="text" 
            placeholder="Search some gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
