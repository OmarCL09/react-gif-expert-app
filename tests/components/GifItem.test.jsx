import { fireEvent, render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests in GifItem component', () => { 

    const title = 'Chainsaw Man';
    const url = 'http://locura.com/diablossenorita.jpg';

    test('should render a gif', () => { 
        const { container } = render(<GifItem title={title} url={url}/>);
        expect( container ).toMatchSnapshot();
    })

    test('debe mostrar la imagen con url y alt indicado', () => { 
        const { container } = render(<GifItem title={ title } url={ url } />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    }) 

    test('debe de mostrar el titulo del componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
}) 