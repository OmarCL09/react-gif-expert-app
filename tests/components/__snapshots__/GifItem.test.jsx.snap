// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifItem component should render a gif 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Chainsaw Man"
      src="http://locura.com/diablossenorita.jpg"
    />
    <p>
      Chainsaw Man
    </p>
  </div>
</div>
`;
